app-next/components/VolunteerDetail/VolunteerDetail.jsx

```diff
@@ -1,89 +1,78 @@
 "use client";
-
-import React, { useState, useEffect } from "react";
+import React, { useEffect, useState } from "react";
 import styles from "./VolunteerDetail.module.css";
-import { useRouter } from "next/navigation";
-
-const mockVolunteers = [
-  {
-    id: 1,
-    name: "Anna Smith",
-    email: "anna@example.com",
-    phone_nr: "12345678",
-    address: "Copenhagen",
-    photo: "/images/volunteer1.jpg",
-    services: ["Shopping", "Cooking"],
-    gender: "Female",
-    routine: "Morning",
-    availability: "Weekdays, 9am - 1pm",
-  },
-  {
-    id: 2,
-    name: "John Doe",
-    email: "john@example.com",
-    phone_nr: "87654321",
-    address: "Aarhus",
-    photo: "/images/volunteer2.jpg",
-    services: ["Gardening", "Companionship"],
-    gender: "Male",
-    routine: "Evening",
-    availability: "Weekends, 4pm - 8pm",
-  },
-];
 
 const VolunteerDetail = ({ volunteerId }) => {
   const [volunteer, setVolunteer] = useState(null);
-  const router = useRouter();
+  const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    const v = mockVolunteers.find((v) => v.id === Number(volunteerId));
-    setVolunteer(v);
+    const fetchVolunteer = async () => {
+      try {
+        const res = await fetch(
+          `http://localhost:3001/api/volunteers/${volunteerId}`
```

Comment:
We shouldn't use `http://localhost:3001` directly, let's do it though env variable